elements/howto-accordion/howto-accordion.js

```diff
@@ -0,0 +1,533 @@
+          // their end position.
+          animatedChildren.forEach(child => {
+            child.style.transform = `translateY(${endOffset}px)`;
+            child.classList.add('animating');
+          });
+          // Wait for the transition to end.
+          return transitionEndPromise(panel);
+        })
+        .then(_ => {
+          // Clean up all the temporary styles
+          animatedChildren.forEach(child => {
+            child.style.transform = '';
+            child.classList.remove('animating');
+          });
+          children.forEach(child => {
+            child.style.position = '';
+            child.style.zIndex = '';
+          });
+          this.style.overflow = '';
+          this.classList.remove('animating');
+        });
+    }
+  }
+  window.customElements.define('howto-accordion', HowtoAccordion);
+
+  /**
+   * `HowtoAccordionHeading` is the element for the headings in the accordion.
+   * Accordion to the WAI ARIA Best Practices, each heading needs to wrap a
+   * `<button>`. This element dispatches a `howto-accordion-toggle` event when
+   * it is supposed to expand.
+   *
+   * Click the button or pressing space or enter while the button has focus
+   * will expand the button. Changing the `expand` attribute or property will
+   * also cause the heading to expand.
+   */
+  class HowtoAccordionHeading extends HTMLElement {
+    // The element reacts to changes to the `expanded` attribute.
+    static get observedAttributes() {
+      return ['expanded'];
+    }
+
+    constructor() {
+      super();
+
+      // Binding event handlers to `this` ensures that `this` inside the event
+      // handler will always be the `<howto-accordion-heading>`, even the
+      // handler is hooked up to other elements.
+      this._onClick = this._onClick.bind(this);
+    }
+
+    /**
+     * `connectedCallback()` sets up the role, event handler and initial state.
+     */
+    connectedCallback() {
+      if(!this.hasAttribute('role'))
+        this.setAttribute('role', 'heading');
+      const button = this.querySelector('button');
+      if(!button)
+        throw new Error('Missing button inside accordion heading');
+      button.addEventListener('click', this._onClick);
+    }
+
+    /**
+     * `disconnectedCallback` cleans up the event handlers set up in
+     * `connectedCallback`.
+     */
+    disconnectedCallback() {
+      this.querySelector('button').removeEventListener('click', this._onClick);
+    }
+
+    /**
+     * `attributeChangedCallback` processes changes to the `expanded` attribute.
+     */
+    attributeChangedCallback(name) {
+      // `expanded` is a boolean attribute it is either set or not set. The
+      // actual value is irrelevant.
+      const value = this.hasAttribute('expanded');
+
+      // Dispatch an event that signals a request to expand to the
+      // `<howto-accordion>` element.
+      this.dispatchEvent(
+        new CustomEvent('howto-change', {
```

Comment:
I don't think it's wrong necessarily. I just used `change` because I noticed I got freebie Angular directive support with ngModel when I did that. But this isn't a form element so it probably wouldn't use ngModel anyway.

`change` seems ok to me. At least if no better alternative presents itself.